pages/Messages/Validation.md

```diff
@@ -0,0 +1,173 @@
+
+#### Encrypted Messages
+
+In the case of encrypted messages, if the validating user does not possess the necessary keys to decrypt a message, no validation is possible and the message should be treated as valid, i.e. included in a batch, unless proven otherwise.
+Clients and end-users who also do not possess the necessary keys should ignore these messages.
+
+Optionally, archivist implementations may choose to invalidate encrypted messages for which the content is obviously incorrect, i.e. the encrypted byte array isn't long enough to be a valid message, however no standard is defined for this behavior and caution should be used to avoid incorrectly invalidating messages.
+
+If the validating user does possess the necessary keys, the message should decrypted and validated as if it were a non-encrypted message.
+
+#### Non-Encrypted Messages
+
+Validating correctness for a non-encrypted message will consist of verifying that each field in the message is defined and meets the formatting rules listed in the subsections below for the given field type.
+
+##### Addresses
+
+1. Address fields must be exactly 20 bytes.
+
+##### Emoji
+
+1. Emoji fields must not be empty.
+1. Emoji fields must be less than 16 bytes.
+1. Emoji fields must only consist of [Unicode points](https://unicode.org/standard/standard.html) from `U+2000` to `U+2BFF`, from `U+E000` to `U+FFFF`, or from `U+1F000` to `U+10FFFF`.
+
+For example, all of the following should be considered valid emoji:
+
+```
+"😀", "🤌🏼", "👩🏻‍🎤", "🧑🏿‍🏫", "🏳️‍🌈", "🏳️‍⚧️", "⚛︎", "🃑", "♻︎"
+```
+
+Additionally, none of the following should be considered valid:
+
+```
+"F", ":custom-emoji:", "<custom-emoji>", "ᚱ", "ᘐ", "״"
+```
+
+##### Hashes
+
+1. Hash fields must be exactly 32 bytes.
+
+##### Message Ids
+
+1. Message Identifier fields must meet all standards defined in the [Message Identifiers](/Messages/Identifiers) specification.
+
+##### Type Enumerators
+
+1. Type Enumerator fields must be a valid enumerator value as defined in the [Message Overview](/Messages/Overview).
+
+##### URI Fields
+
+1. URI fields must include meet all standards defined in [RFC3986](http://www.ietf.org/rfc/rfc3986.txt).
+1. URI fields must not refer to localhost or any reserved IP addresses as defined in [RFC6890](https://datatracker.ietf.org/doc/html/rfc6890)
+1. URI fields must use either a `http` or `https` protocol. Support for other protocols may be added in the future.
+1. URI fields must not exceed 1024 bytes.
+
+### Announcement Authenticity
+
+Validation of announcement authenticity will consist of verifying the message's signature using the appropriate public key for the user listed in the `fromAddress` of the message and the [Secp256k1](https://en.bitcoin.it/wiki/Secp256k1) algorithm, as defined in the [Message Signatures](/Messages/Signatures) specification.
```

Comment:
I think we should get more detailed here.

`fromAddress` I think is going to translate into fromId (being the DSNP Id). So the steps to are:

1. Resolve the fromId to a identity contract
2. using ecrecovery get the public key from the signature
3. Test the public key against the identity contract with IDelegation.isAuthorizedTo